webdriver/tests/bidi/browsing_context/close/prompt_unload.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+import asyncio
+
+from webdriver.bidi.modules.script import ContextTarget
+
+pytestmark = pytest.mark.asyncio
+
+CONTEXT_DESTROYED_EVENT = "browsingContext.contextDestroyed"
+USER_PROMPT_OPENED_EVENT = "browsingContext.userPromptOpened"
+
+
+@pytest.mark.parametrize("type_hint", ["window", "tab"])
+@pytest.mark.parametrize("prompt_unload", [None, False])
+async def test_prompt_unload_not_triggering_dialog(bidi_session, subscribe_events, url, type_hint, prompt_unload):
+
+    new_context = await bidi_session.browsing_context.create(type_hint=type_hint)
+
+    page_beforeunload = url(
+        "/webdriver/tests/support/html/beforeunload.html")
+
+    # Set up event listener to make sure the "beforeunload" event is not emitted
+    await subscribe_events([USER_PROMPT_OPENED_EVENT])
+    # Track all received browsingContext.userPromptOpened events in the events array
+    events = []
+
+    async def on_event(_, data):
+        events.append(data)
+    remove_listener = bidi_session.add_event_listener(
+        USER_PROMPT_OPENED_EVENT, on_event)
+
+    await bidi_session.browsing_context.navigate(context=new_context["context"], url=page_beforeunload, wait="complete")
+
+    # We need to interact with the page to trigger the beforeunload event.
+    # https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event#usage_notes
+    await bidi_session.script.evaluate(
+        expression="document.querySelector('input').click()",
+        target=ContextTarget(new_context["context"]),
+        await_promise=True,
+        user_activation=True)
+
+    await bidi_session.browsing_context.close(context=new_context["context"], prompt_unload=prompt_unload)
+
+    assert events == []
+
+    remove_listener()
+
+
+@pytest.mark.parametrize("type_hint", ["window", "tab"])
+async def test_prompt_unload_triggering_dialog(bidi_session, url, subscribe_events, wait_for_event, type_hint):
+    page_beforeunload = url(
+        "/webdriver/tests/support/html/beforeunload.html")
+
+    new_context = await bidi_session.browsing_context.create(type_hint=type_hint)
+
+    # Set up event listener to make sure the "beforeunload" event is not emitted
+    await subscribe_events([USER_PROMPT_OPENED_EVENT, CONTEXT_DESTROYED_EVENT])
+    user_prompt_opened = wait_for_event(USER_PROMPT_OPENED_EVENT)
+
+    # Track all received browsingContext.contextDestroyed events in the events array
+    events = []
+
+    async def on_event(_, data):
+        if data["type"] == CONTEXT_DESTROYED_EVENT:
+            events.append(data)
```

Comment:
```suggestion
        events.append(data)
```

Since we are only listening to CONTEXT_DESTROYED_EVENT, we shouldn't need the `if` here?

Reply:
Not true we listen to `USER_PROMPT_OPENED_EVENT` and `CONTEXT_DESTROYED_EVENT`. [Line 56](https://github.com/web-platform-tests/wpt/pull/44004/files/8049d4bc5bf7c2c6a67e1d8379eee1c14dbc0e9c#diff-531992b4df60c2c5caf6a167577064c5f1ed4933e8c1b5988f528dc3e7bbdab7R56)